### PR TITLE
Improve release process, ensure proper docker tags, ensure CI jobs on…

### PR DESCRIPTION
**Pull request by devinrsmith**
… release branches.

These changes are part of the history leading up to tag v0.1.0.

### CODE REVIEW

File: .github/workflows/check-ci.yml
Comment (mofojed) [inline .github/workflows/check-ci.yml:5]: Shouldn't we run checks on PRs to release branches as well? They should pass tests before merging, even if it's being cherry-picked from main which has a passing test (just because it passes on main doesn't mean it passes on release/*)

File: .github/workflows/build-ci.yml
Comment (mofojed) [inline .github/workflows/build-ci.yml:24]: I don't know if this is possible, but `startsWith(github.ref, 'refs/heads/release/v')` is used in a bunch of places, can that be initialized at the start and re-used? Maybe as an env variable?

File: .github/workflows/build-ci.yml
Comment (mofojed) [inline .github/workflows/build-ci.yml:113]: Duplicate `priority` attribute
```suggestion
            type=raw,priority=950,enable=${{ startsWith(github.ref, 'refs/heads/release/v') }},value=${{ steps.semver.outputs.semver }}
```

File: .github/workflows/build-ci.yml
Comment (mofojed) [inline .github/workflows/build-ci.yml:114]: If I'm reading this right, won't this add the tag `latest` when doing a push back to any release branch? If we have `v2.0.0`, and `v1.5.0`, and then we patch 1.5 to make `v1.5.1`, won't that get tagged `latest` when `v2.0.0` should still be `latest`?

**Review by mofojed — CHANGES_REQUESTED** [review]

File: .github/workflows/check-ci.yml
Comment (devinrsmith) [reply .github/workflows/check-ci.yml:5]: We don't have long-lived release branches, and aren't really supporting PRs into short-lived release branches. We should be empowered to re-write history on our short-lived release branches if necessary; so just pushes to them should be OK IMO. It's really only the release manager that should be interacting directly w/ the release branch for now.

File: .github/workflows/build-ci.yml
Comment (devinrsmith) [reply .github/workflows/build-ci.yml:24]: There's a lot of workflow yml duplication (ie, the java version too) - I don't think there's a "proper" way to do de-duplication at the configuration level (GH yaml doesn't support yaml anchors :/). Changing the environment *may* work, but is technically fragile since the values can change at runtime. Will investigate.

File: .github/workflows/build-ci.yml
Comment (devinrsmith) [reply .github/workflows/build-ci.yml:114]: Yes - that's the problem w/ the `latest` tag in general - I've noticed a lot of projects fall into this trap.

https://github.com/docker/metadata-action/issues/80

It can be solved by guarding the latest tag against a specific ref - I could update this to

```
             type=raw,enable=${{ startsWith(github.ref, 'refs/heads/release/v0.2') }},value=latest 
```
, and we just need to make sure when we do version bump, we also update this.

**Review by mofojed — APPROVED** [review]

**Review by jcferretti — APPROVED** [review]

**Pull request merged**
